examples/flutter_gallery/lib/demo/pan_and_zoom_demo_transform_interaction.dart

```diff
@@ -0,0 +1,469 @@
+    // always aligned with the viewport.
+    this.visibleRect,
+    // Initial values for the transform can be provided
+    this.initialTranslation,
+    this.initialScale,
+    this.initialRotation,
+    // Any and all of the possible transformations can be disabled.
+    this.disableTranslation = false,
+    this.disableScale = false,
+    this.disableRotation = false,
+    // If set to true, this widget will animate back to its initial transform
+    // and call onResetEnd when done. When utilizing reset, onResetEnd should
+    // also be implemented, and it should set reset to false when called.
+    this.reset = false,
+    // Access to event callbacks from GestureDetector. Called with untransformed
+    // coordinates in an Offset.
+    this.onTapDown,
+    this.onTapUp,
+    this.onTap,
+    this.onTapCancel,
+    this.onDoubleTap,
+    this.onLongPress,
+    this.onLongPressUp,
+    this.onVerticalDragDown,
+    this.onVerticalDragStart,
+    this.onVerticalDragUpdate,
+    this.onVerticalDragEnd,
+    this.onVerticalDragCancel,
+    this.onHorizontalDragDown,
+    this.onHorizontalDragStart,
+    this.onHorizontalDragUpdate,
+    this.onHorizontalDragEnd,
+    this.onHorizontalDragCancel,
+    this.onPanDown,
+    this.onPanStart,
+    this.onPanUpdate,
+    this.onPanEnd,
+    this.onPanCancel,
+    this.onResetEnd,
+    this.onScaleStart,
+    this.onScaleUpdate,
+    this.onScaleEnd,
+  }) :
+    assert(!reset || onResetEnd != null, 'Must implement onResetEnd to use reset.');
+
+  final Widget child;
+  final Size size;
+  final bool reset;
+  final Function onTapDown;
+  final Function onTapUp;
+  final Function onTap;
+  final Function onTapCancel;
+  final Function onDoubleTap;
+  final Function onLongPress;
+  final Function onLongPressUp;
+  final Function onVerticalDragDown;
+  final Function onVerticalDragStart;
+  final Function onVerticalDragUpdate;
+  final Function onVerticalDragEnd;
+  final Function onVerticalDragCancel;
+  final Function onHorizontalDragDown;
+  final Function onHorizontalDragStart;
+  final Function onHorizontalDragUpdate;
+  final Function onHorizontalDragEnd;
+  final Function onHorizontalDragCancel;
+  final Function onPanDown;
+  final Function onPanStart;
+  final Function onPanUpdate;
+  final Function onPanEnd;
+  final Function onPanCancel;
+  final Function onResetEnd;
+  final Function onScaleStart;
+  final Function onScaleUpdate;
+  final Function onScaleEnd;
+  final double maxScale;
+  final double minScale;
+  final Rect visibleRect;
+  final bool disableTranslation;
+  final bool disableScale;
+  final bool disableRotation;
+  final Offset initialTranslation;
+  final double initialScale;
+  final double initialRotation;
+
+  @override TransformInteractionState createState() => TransformInteractionState();
+}
+
+enum GestureType {
+  translate,
+  scale,
+  rotate,
+}
+
+class TransformInteractionState extends State<TransformInteraction> with TickerProviderStateMixin {
```

Comment:
Why is this public? Who is expected to call its APIs? Document them

Reply:
I just realized that I made this public so that I can access it in a unit test.  Is that valid idea or is there a better way?